iOS/Flows/Conversation/Cells/ConversationMessagesCell.swift

```diff
@@ -37,15 +38,30 @@ class ConversationMessagesCell: UICollectionViewCell {
     private var state: ConversationUIState = .read
 
     /// The conversation containing all the messages.
-    var conversation: MessageSequence?
+    var conversation: Conversation? {
+        return self.conversationController?.conversation
+    }
+    private(set) var conversationController: ConversationController?
```

Comment:
I ultimately found it easier to put a conversation controller on the cell itself and have the cell subscribe for updates. That way I could ensure that "hasLoadedMessages" status stayed consistent.